src/v0/destinations/hs/util.js

```diff
@@ -159,6 +158,32 @@ const getTransformedJSON = async (message, destination, propertyMap) => {
           date.setUTCHours(0, 0, 0, 0);
           propValue = date.getTime();
         }
+
+        // Hub spot data type validations 
+        if (propertyMap[hsSupportedKey] === 'string' && typeof propValue !== 'string') {
+          if (typeof propValue === 'object') {
+            propValue = JSON.stringify(propValue);
+          } else {
+            propValue = propValue.toString();
+          }
+        }
+
+        if (propertyMap[hsSupportedKey] === 'bool' && typeof propValue === 'object') {
+          throw new InstrumentationError(
+            `Property ${traitsKey} data type ${typeof propValue} is not matching with Hubspot property data type ${
+              propertyMap[hsSupportedKey]
+            }`,
+          );
+        }
+
+        if (propertyMap[hsSupportedKey] === 'number' && typeof propValue !== 'number') {
+          throw new InstrumentationError(
+            `Property ${traitsKey} data type ${typeof propValue} is not matching with Hubspot property data type ${
+              propertyMap[hsSupportedKey]
+            }`,
+          );
```

Comment:
thinking out loud, can we set a data type which is an array, object?

Reply:
Hubspot only has below data types
- string
- enumeration
- date
- number
- bool

verified by creating property using all possible options but did not found object or array of object data type